tests/test_utils_url.py

```diff
@@ -123,14 +123,12 @@ def test_spaces(self):
         self.assertEqual(canonicalize_url("http://www.example.com/do?q=a%20space&a=1"),
                                           "http://www.example.com/do?a=1&q=a+space")
 
-    @unittest.skipUnless(six.PY2, "TODO")
     def test_normalize_percent_encoding_in_paths(self):
         self.assertEqual(canonicalize_url("http://www.example.com/a%a3do"),
-                                          "http://www.example.com/a%A3do"),
+                                          "http://www.example.com/a%A3do")
 
-    @unittest.skipUnless(six.PY2, "TODO")
     def test_normalize_percent_encoding_in_query_arguments(self):
-        self.assertEqual(canonicalize_url("http://www.example.com/do?k=b%a3"),
+        self.assertEqual(canonicalize_url("http://www.example.com/do?k=b%a3", encoding='latin1'),
```

Comment:
What happens if a wrong encoding is passed?
canonicalize_url is used in dupefilters; a default encoding is used there instead of the response encoding. What happens if this default encoding is not correct? Can we get duplicates which are not really duplicates? Can we get an exception?